### DISCUSSION TITLE
Placeholder API (hear me out)

### DISCUSSION THREAD

**Discussion by khoingt** [GriefPrevention/GriefPrevention#1548] [Ideas]
**As** a player
**I want** one complex placeholder (I'll explain below)
**Because** I want the placeholder to be more flexible and I want to give the plugin a cosmetic touch :>

This is what I mean:
- Placeholder that changes when ever a player walks into a claimed/un-claimed land. The placeholder could be %land_claim_who%. This is how it might be presented (and yes, i'm aware that there's a similar placeholder to this, but it's not flexible):
   + For un-claimed lands, the placeholder will show the following: `Unclaimed Land` (customizable)
   + For claimed lands, the placeholder will show the following: `{PLAYER} Claim` (customizable)

**Comment by bigpresh** [comment]
Sadly, I think this would be quite expensive, performance-wise, as you'd have to listen to a whole load of PlayerMoveEvent events and do a claim lookup every time to see if the player is now in a claim.  There have been similar features (e.g. claim entry/exit messages) which have been avoided for the same reason in the past.

It may be possible to implement as an addon?

**Reply by khoingt** [reply]
an add-on would've be nice :>
also, can i change how the placeholder %griefprevention_currentclaim_ownername% looks like?

**Comment by Jikoo** [comment]
The PAPI extension is not maintained here, and I'm pretty sure the maintainer does not follow this repo at all. You'll have much better luck asking over at https://github.com/pixar02/Expansions

**Reply by khoingt** [reply]
ty!
